src/test/scala/com/codacy/parsers/CloverParserTest.scala

```diff
@@ -0,0 +1,60 @@
+package com.codacy.parsers
+
+import java.io.File
+
+import com.codacy.parsers.implementation.CloverParser
+import org.scalatest.{BeforeAndAfterAll, EitherValues, Matchers, WordSpec}
+
+class CloverParserTest extends WordSpec with BeforeAndAfterAll with Matchers with EitherValues {
+  "CloverParser" should {
+    "identify if report is invalid" in {
+      // use some coverage file that does not follow the Clover format
+      val reader = CloverParser.parse(
+        new File("."),
+        new File("src/test/resources/test_cobertura.xml"))
+
+      reader.isLeft shouldBe true
+    }
+
+    "identify if report is valid" in {
+      val reader = CloverParser.parse(
+        new File("."),
+        new File("src/test/resources/test_clover.xml"))
+      reader.isRight shouldBe true
+    }
+
+    "return a valid report" in {
+      val reader = CloverParser.parse(
+        new File("/home/codacy-php/"),
+        new File("src/test/resources/test_clover.xml"))
+
+
+      val report = reader.right.value
+      report.total shouldBe 38
+      // 5 files used in the initial test
+      report.fileReports.length shouldBe 5
+
+      // coverage percentage for file Parser.php
+      report.fileReports.head.total shouldBe 33
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://app.codacy.com/gh/codacy/coverage-parser/pullRequest?prid=4898850)